src/shared/dissect-image.c

```diff
@@ -1059,6 +1053,9 @@ int dissect_image(
                 if (verity->designator >= 0 && !m->partitions[verity->designator].found)
                         return -EADDRNOTAVAIL;
 
+                bool have_verity_sig_partition =
+                        m->partitions[verity->designator == PARTITION_USR ? PARTITION_USR_VERITY_SIG : PARTITION_ROOT_VERITY_SIG].found;
```

Comment:
Maybe, better to use `PARTITION_VERITY_SIG_OF()`?

Reply:
The rest of the code seems to assume that `verity->designator` can be invalid sometimes, so using `PARTITION_VERITY_SIG_OF()` could lead to different results.

This needs some greater refactoring to make sure that `verity->designator` can never be null first I think (always default to `PARTITION_ROOT`) but I'd like to leave that for a separate PR.

Reply:
Sure, OK.